Zend/Optimizer/optimize_func_calls.c

```diff
@@ -117,6 +117,13 @@ static void zend_try_inline_call(zend_op_array *op_array, zend_op *fcall, zend_o
 				} while (i < func->op_array.num_args);
 			}
 
+			for (i = 0; i < op_array->last; i++) {
+				// Don't inline calls with explicit default arguments.
+				if (op_array->opcodes[i].opcode == ZEND_FETCH_DEFAULT_ARG) {
+					return;
+				}
+			}
```

Comment:
As mentioned off-list, this should check only for opcodes within the current call (between `fcall` and `opline`), not the entire function. It would be nice to also skip nested functions, but that will require listing all init/do opcodes.

Reply:
Thanks for the note. I am still trying to figure out how to do this.

Reply:
Store fcall in a local var, loop, and bump until reaching opline.

Reply:
There is code for that in PHP already somewhere, and now also in Xdebug: https://github.com/xdebug/xdebug/commit/551b203b27192e15dadf51577e490f03c2a3dfe2#diff-846d4897469701b1d805f3574922dc47b632fd189087a2589b2e01e848585a8bR378 — that should probably become a PHP API function.

Reply:
I'll be honest, I still have no idea what to do here, and no idea how the Xdebug code relates to this. Anyway, I'll deal with this if the feature gets accepted.

Reply:
```c
for (zend_op *op = fcall + 1; op != opline; op++) {
	// Don't inline calls with explicit default arguments.
	if (op->opcode == ZEND_FETCH_DEFAULT_ARG) {
		return;
	}
}
```

Reply:
Thanks. Clearly I need to understand more about what oparrays, oplines and opcodes really are, and how they relate to each other.